controllers/author-controller.js

```diff
@@ -0,0 +1,210 @@
+var Author = require("../models/author");
+var Book = require("../models/book");
+
+var async = require("async");
+
+var { body, validationResult } = require("express-validator");
+
+// Display list of all Authors.
+exports.author_list = function (req, res) {
+  Author.find()
+    .sort([["family_name", "ascending"]])
+    .exec(function (err, list_authors) {
+      if (err) {
+        return next(err);
+      }
+      //Successful, so render
+      res.render("author-list", {
+        title: "Author List",
+        author_list: list_authors,
+      });
+    });
+};
+
+// Display detail page for a specific Author.
+exports.author_detail = function (req, res) {
+  async.parallel(
+    {
+      author: function (callback) {
+        Author.findById(req.params.id).exec(callback);
+      },
+      authors_books: function (callback) {
+        Book.find({ author: req.params.id }, "title summary").exec(callback);
+      },
+    },
+    function (err, results) {
+      if (err) {
+        return next(err);
+      } // Error in API usage.
+      if (results.author == null) {
+        // No results.
+        var err = new Error("Author not found");
+        err.status = 404;
+        return next(err);
+      }
+      // Successful, so render.
+      res.render("author-detail", {
+        title: "Author Detail",
+        author: results.author,
+        author_books: results.authors_books,
+      });
+    }
+  );
+};
+
+// Display Author create form on GET.
+exports.author_create_get = function (req, res) {
+  res.render("author-form", { title: "Create Author" });
+};
+
+// Handle Author create on POST.
+exports.author_create_post = [
+  // Validate and sanitise fields.
```

Comment:
Please move the validation part to a new file, make a validators folder in the root.

Reply:
Run the validator before the `author_create_post` is called.

Reply:
ok